### PR TITLE
Relay the moderator's reason for `!superstarify` to the user

### PR DESCRIPTION
**Pull request by swfarnsworth**
This PR closes issue #1224.

Previously, when a moderator used the `!superstarify` command, the reason they gave only gets recorded in the user's infraction history and is not relayed to the user in the direct message sent by the bot. This PR changes the message the user is sent to fix this.

However, we're discussing if the special embed that gets posted in the channel where the command is issued is actually appropriate.

image

### CODE REVIEW

**Comment by swfarnsworth** [comment]
Here's an alternative message (I have not committed or pushed this yet):

image

This is the DM that they get:
image

This is what the mod would see in the user's infraction history:
image

**Comment by swfarnsworth** [comment]
Per @mbaruh's suggestion, I made another change whereby the "Additional details: " portion of the message does not appear in the embed displayed in the channel (since it's just restating what the mod wrote in the command), whereas the message still appears in the message sent to the user. The messages are otherwise the same.

**Comment by swfarnsworth** [comment]
image

Per an internal discussion, we'll retain the original wording of the embed, but the title will still be "Superstarify!" instead of "Congratulations!"

File: bot/exts/moderation/infraction/superstarify.py [hunk 1]
```diff
@@ -128,16 +128,16 @@ async def superstarify(
 
         Alternatively, an ISO 8601 timestamp can be provided for the duration.
 
-        An optional reason can be provided. If no reason is given, the original name will be shown
-        in a generated reason.
+        An optional reason can be provided, which would be added to a message stating their old nickname
+        and linking to the nickname policy.
```
Comment (HassanAbouelela) [inline bot/exts/moderation/infraction/superstarify.py:132]: Was there an official consensus on echoing back the user's name, or is that not a concern?

File: bot/exts/moderation/infraction/superstarify.py [hunk 2]
```diff
@@ -152,15 +152,26 @@ async def action() -> None:
...
         """).strip()
 
+        formatted_reason = f'**Additional details:** {reason}\n\n' if reason else ''
+
+        embed_reason = (
+            f"Your previous nickname, **{old_nick}**, "
+            f"didn't comply with our nickname policy. "
+            f"Your new nickname will be **{forced_nick}**.\n\n"
+            f"{formatted_reason}"
+            f"You will be unable to change your nickname until **{expiry_str}**. "
```
Comment (HassanAbouelela) [inline bot/exts/moderation/infraction/superstarify.py:167]: The expiry time is duplicated in two different spots in the direct message embed ([L115](https://github.com/python-discord/bot/blob/c3a927569da782c24299c8ae75df28ae6cd3f2ba/bot/exts/moderation/infraction/_scheduler.py#L115)). Can you remove this from the main embed, and only add it to the message sent to the invocation context?

File: bot/exts/moderation/infraction/superstarify.py [hunk 3]
```diff
@@ -171,18 +182,11 @@ async def action() -> None:
             embed = Embed(
                 title="Congratulations!",
```
Comment (HassanAbouelela) [inline bot/exts/moderation/infraction/superstarify.py:183]: Don't forget to update this, based on the poll results.

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]

File: bot/exts/moderation/infraction/superstarify.py (hunk 1 above)
Comment (mbaruh) [reply bot/exts/moderation/infraction/superstarify.py:132]: This is referring to what will appear in the infraction history.

File: bot/exts/moderation/infraction/superstarify.py (hunk 2 above)
Comment (mbaruh) [reply bot/exts/moderation/infraction/superstarify.py:167]: I think it's important to specify that they will be unable to change the nickname until the infraction expires, and in that case duplicating the date doesn't seem too bad.

**Comment by swfarnsworth** [comment]
The public embed is always the same:
image

The infraction message depends on whether or not the mod gave a reason:
image

image

The message in the infraction history is unchanged from the previous set of screenshots.

File: bot/exts/moderation/infraction/superstarify.py (hunk 1 above)
Comment (HassanAbouelela) [reply bot/exts/moderation/infraction/superstarify.py:132]: Sure, but it does appear in the infraction history, DMs, and server invocation channel. I am curious if we want to be sending the old names as part of the (public-facing) embeds or not. It was a point touched upon during discussions, but not fully explored.

File: bot/exts/moderation/infraction/superstarify.py (hunk 2 above)
Comment (HassanAbouelela) [reply bot/exts/moderation/infraction/superstarify.py:165]: We can still mention that you can't change your name while this is in effect, without adding the date, but at that point, we're splitting hairs. If you're fine with the duplication, then I won't complain.

File: bot/exts/moderation/infraction/superstarify.py [hunk 4]
```diff
@@ -152,37 +152,38 @@ async def action() -> None:
...
 
+        user_message = (
+            f"Your previous nickname, **{old_nick}**, "
+            f"was so bad that we have decided to change it. "
+            f"Your new nickname will be **{forced_nick}**.\n\n"
+            "{reason}"
+            f"You will be unable to change your nickname until **{expiry_str}**. "
+            "If you're confused by this, please read our "
+            f"[official nickname policy]({NICKNAME_POLICY_URL})."
+        ).format
```
Comment (Den4200) [inline bot/exts/moderation/infraction/superstarify.py:168]: This feels a bit unconventional and odd, but it's quite smart.

**Review by Den4200 — APPROVED** [review]
Looks and works great!

File: bot/exts/moderation/infraction/superstarify.py (hunk 1 above)
Comment (mbaruh) [reply bot/exts/moderation/infraction/superstarify.py:132]: infraction history and DM - yes, we want to know what the old nick was for future reference, and I don't see a reason for it to not be DMed.
For the public embed, from what I gathered people didn't seem to think it was an major issue, and most starts are applied form private channels anyway. Also in the vote it says to keep the content, and I can only assume it includes the old nick.

**Review by mbaruh — APPROVED** [review]

**Review by HassanAbouelela — APPROVED** [review]

**Pull request merged**